assets/javascripts/initializers/discourse-math-katex.js

```diff
@@ -38,9 +38,19 @@ function katex($elem) {
 
   if (mathElems.length > 0) {
     ensureKaTeX().then(() => {
-      // enable persistent macros: https://katex.org/docs/api.html#persistent-macros
-      const macros = {};
-      mathElems.each((idx, elem) => decorate(elem, macros));
+      // enable persistent macros with are disabled by default: https://katex.org/docs/api.html#persistent-macros
+      // also enable equation labelling and referencing which are disabled by default
+      // both of these are enabled in mathjax by default, so now the katex implementation is (more) mathjax compatible
+      const katexOpts = {
+        trust: (context) => ['\\htmlId', '\\href'].includes(context.command),
+        macros: {
+          "\\eqref": "\\href{###1}{(\\text{#1})}",
+          "\\ref": "\\href{###1}{\\text{#1}}",
+          "\\label": "\\htmlId{#1}{}"
+        },
+        displayMode: false,
```

Comment:
We can probably set the displayMode value based on the element here. This should work (untested)

```suggestion
        displayMode: elem.tagName === "DIV",
```

and we don't need to do the same calculation in the decorator.